README.Rmd

```diff
@@ -14,57 +13,146 @@ knitr::opts_chunk$set(
 
-On Mac
+  ```{cmd eval=F}
+  python --version
+  Python3 --version
+  ```
 
-```{r eval=F}
-install.packages("synapser", repos=c("http://ran.synapse.org", "http://cran.fhcrc.org"), type="mac.binary")
-```
+- If you don't then go ahead and install it with the installer. Go the the python's official site
+[here](https://www.python.org/downloads/mac-osx/).
 
-On Windows:
+- Now restart the terminal and check again with both commands python —version
 
-```{r eval=F}
-install.packages("synapser", repos=c("http://ran.synapse.org", "http://cran.fhcrc.org"), type="win.binary")
-```
+  ```{cmd eval=F}
+  Python3 --version
+  ```
+
+- `Or` use to `install last version`
+
+  ```{cmd eval=F}
+  brew install python3 && cp /usr/local/bin/python3 /usr/local/bin/python
+  ```
+
+## How to Upgrade Python on Linux
+
+- Add the repository and update
+
+  ```{cmd eval=F}
+  sudo add-apt-repository ppa:deadsnakes/ppa
+  sudo apt-get update
+  ```
+
+- Update the package list
+
+  ```{cmd eval=F}
+  apt-get update
+  ```
+
+- Verify the updated Python packages list
+
+    ```{cmd eval=F}
+    $apt list | grep python3.10
+    ```
+
+- Install the Python 3.10 package using apt-get
+
+    ```{cmd eval=F}
+    sudo apt-get install python3.10
+    ```
+
+- Add Python 3.8 & Python 3.10 to update-alternatives
+
+  ```{cmd eval=F}
+  sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.8 1
+  sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.10 2
+  ```
+
+- Update Python 3 for point to Python 3.10
+
+  ```{cmd eval=F}
+  sudo update-alternatives --config python3
+  ```
+
+## How to Upgrade R on Windows
+
+- Verify R version
+  - Open RStudio > At the top of the Console you will see session info > The first line tells you which version of R you are using.
+  - `or` write in console >'R.version.string' to print out the R version.
+  - go to Tools > Check for Package Updates. If there’s an update available for tidyverse, install it.
+
+## for Windows
+
+- To update R on Windows, try using the package installer (only for Windows).
+- Got to Tools (at the top) > Check for package updates. If tidyverse shows up on the list, select it, then click “Install Updates.”
+
+## for Mac
+
+- Go to
+[here](https://cloud.r-project.org/bin/macosx/).
+- Click the link you need to update R pkg
+- When the file finishes downloading, double-click to install. You should be able to click “Next” to all dialogs to finish the installation.
+- From within RStudio, go to Help > Check for Updates to install newer version of RStudio (if available, optional).
+- To update packages, go to Tools > Check for Package Updates. If updates are available, select All (or just tidyverse), and click Install Updates.
+
+## To build and install follow these steps
+
+- build package from R console:
+
+  ```{r eval=F}
+  R CMD BUILD .
```

Comment:
I would point people to this: https://remotes.r-lib.org/reference/install_local.html instead

Reply:
Would you prefer `install_local`, or `install_github`?
```
install.packages(c('remotes', 'reticulate'))
remotes::install_github(repo='https://github.com/Sage-Bionetworks/synapser/tree/master', ref='reticulate', force=T)
```